extensions/Worker.Extensions.Tables/src/TypeConverters/TableEntityEnumerableConverter.cs

```diff
@@ -0,0 +1,103 @@
+{
+    /// <summary>
+    /// Converter to bind <see cref="IEnumerable{T}" /> of type <see cref="TableEntity"/> parameters.
+    /// </summary>
+    [SupportsDeferredBinding]
+    [SupportedTargetType(typeof(IEnumerable<TableEntity>))]
+    internal class TableEntityEnumerableConverter : TableConverterBase<IEnumerable<TableEntity>>
+    {
+        public TableEntityEnumerableConverter(IOptionsMonitor<TablesBindingOptions> tableOptions, ILogger<TableEntityEnumerableConverter> logger)
+            : base(tableOptions, logger)
+        {
+        }
+
+        public override async ValueTask<ConversionResult> ConvertAsync(ConverterContext context)
+        {
+            try
+            {
+                if (!CanConvert(context))
+                {
+                    return ConversionResult.Unhandled();
+                }
+
+                var modelBindingData = context?.Source as ModelBindingData;
+                var tableData = GetBindingDataContent(modelBindingData);
+                var result = await ConvertModelBindingData(tableData);
+
+                return ConversionResult.Success(result);
+            }
+            catch (Exception ex)
+            {
+                return ConversionResult.Failed(ex);
+            }
+        }
+
+        private async Task<IEnumerable<TableEntity>> ConvertModelBindingData(TableData content)
+        {
+            if (string.IsNullOrEmpty(content.TableName))
+            {
+                throw new ArgumentNullException(nameof(content.TableName));
+            }
+
+            if (content.RowKey != null && (content.Take > 0 || content.Filter != null))
+            {
+                throw new InvalidOperationException($"Row key {content.RowKey} cannot have a value if {content.Take} or {content.Filter} are defined");
+            }
+
+            return await GetEnumerableTableEntity(content);
+        }
+
+        private async Task<IEnumerable<TableEntity>> GetEnumerableTableEntity(TableData content)
+        {
+            var tableClient = GetTableClient(content.Connection, content.TableName!);
+            string? filter = content.Filter;
+
+            if (!string.IsNullOrEmpty(content.PartitionKey))
+            {
+                var partitionKeyPredicate = TableClient.CreateQueryFilter($"PartitionKey eq {content.PartitionKey}");
+                filter = !string.IsNullOrEmpty(content.Filter) ? $"{partitionKeyPredicate} and {content.Filter}" : partitionKeyPredicate;
+            }
+
+            int? maxPerPage = null;
+            if (content.Take > 0)
+            {
+                maxPerPage = content.Take;
+            }
+
+            int countRemaining = content.Take;
+
+            var entities = tableClient.QueryAsync<TableEntity>(
+                            filter: filter,
+                            maxPerPage: maxPerPage).ConfigureAwait(false);
+
+            List<TableEntity> bindingDataContent = new List<TableEntity>();
+
+            await foreach (var entity in entities)
```

Comment:
Why are we materializing this ourselves instead of just letting the user enumerate themselves? The In-proc extension did this (looks like this code is a copy) because it was typed as JArray. But here the user is binding to an IEnumerable so we could just hand that back to them?

Reply:
Just to clarify, you're saying that the code was copied from [CreateJArray](https://github.com/Azure/azure-sdk-for-net/blob/f738f6a6b3f7f95114cc92fa26ded100907b74de/sdk/tables/Microsoft.Azure.WebJobs.Extensions.Tables/src/TablesExtensionConfigProvider.cs#L135), and in that code they did this because they had to create a JArray. We do not have to do this because `entities` is already IEnumerable so we should be able to just return that?

I agree with you here; I missed this in the original [PR](https://github.com/Azure/azure-functions-dotnet-worker/pull/1506), but returning entities directly here makes the most sense - thanks for catching!

Reply:
Though I wonder if we would still need to convert this to IEnumerable since technically `entities` is actually `ConfiguredCancelableAsyncEnumerable<TableEntity>`. Or if we should change the supported binding type to IAsyncEnumerable

Reply:
@mathewc thoughts on this ^ ? I updated the code but still creating an IList

Reply:
You're pointing out that the entities is actually an `AsyncPageable<T>`? Then perhaps that's what we should be returning - there are methods on there related to paging that the user might care about. When we do a ToList, it ends up enumerating all the pages (possibly multiple service calls), right? Our goal with SDK bindings is to allow the user to bind to SDK natural types. So we need to decide if we want to interject ourselves and force enumerate/convert.

Reply:
Discussed offline, continuing with IEnumerable for now as we need to take into account Take for parity. Future work for IAsyncPageable